addons/project/models/project_task.py

```diff
@@ -630,18 +630,19 @@ def _extract_tags_and_users(self):
         self.display_name, dummy = re.subn(pattern, '', self.display_name)
 
     def _extract_priority(self):
-        self.priority = "1"
-        priority_group = self._get_group_pattern()['priority']
-        self.display_name, dummy = re.subn(priority_group, '', self.display_name)
+        if len(self.display_name) > 1 and self.display_name.endswith('!'):
+            self.priority = "1"
+            self.display_name = self.display_name[:len(self.display_name)-1]
+            self.display_name = self.display_name.strip()
 
     def _get_groups(self):
         return [
             lambda task: task._extract_tags_and_users(),
-            lambda task: task._extract_priority(),
         ]
 
     def _inverse_display_name(self):
         for task in self:
+            task._extract_priority()
```

Comment:
Should the priority be treated independently from the rest, even if the task name starts with a forbidden pattern? For example, the task "#test!" would still have a priority of 1, even though it starts with "#".

Note that the "!" would also not be removed from the name, but if we decide to treat priority separately, this can be easily fixed with:
```diff
            if match:
                for group, extract_data in enumerate(task._get_groups(), start=1):
                    if match.group(group):
                        extract_data(task)
-                task.name = task.display_name.strip()
+           task.name = task.display_name.strip()
```

Reply:
i prefered putting it outside as i consider
#anything! => name = "anything" and priority = 1
a valid transformation, let's see the feedback during the test and when FP test it as he reported it